pkg/cmd/cli/cmd/project.go

```diff
@@ -0,0 +1,66 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	kclientcmd "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+	"github.com/golang/glog"
+	"github.com/openshift/origin/pkg/cmd/cli/config"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/spf13/cobra"
+)
+
+func NewCmdProject(f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "project <project-name>",
+		Short: "switch to another project",
+		Long:  `Switch to another project and set it in the config file`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) != 1 {
+				glog.Fatalf("You need to provide the project name.")
+			}
+
+			projectName := args[0]
+
+			oClient, _, err := f.Clients(cmd)
+			if err != nil {
+				glog.Fatalf("Error getting client: %v", err)
+			}
+
+			// TODO fetch only projects the user has access https://github.com/openshift/origin/pull/971
+			project, err := oClient.Projects().Get(projectName)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					glog.Fatalf("Unable to find a project with name '%v'.", projectName)
+				} else {
+					glog.Fatalf("%v", err)
+				}
+			}
+
+			clientCfg, err := f.OpenShiftClientConfig.ClientConfig()
+			if err != nil {
+				glog.Fatalf("%v", err)
+			}
+
+			configStore, err := config.GetConfigFromDefaultLocations(clientCfg, cmd)
+			if err != nil {
+				glog.Fatalf("%v", err)
+			}
+
+			currentCtx := configStore.Config.Contexts[configStore.Config.CurrentContext]
+
+			currentCtx.Namespace = project.Name
+			configStore.Config.Contexts[configStore.Config.CurrentContext] = currentCtx
```

Comment:
When we discussed this in the path we said that switching projects would create a new context, and that when we switched we would check first for a context with the provided project name.